packages/sdk/src/types.ts

```diff
@@ -60,41 +60,63 @@ export type MakeBuyOrderParams = MakeSellOrderParams;
 export enum MarketplaceName {
   Opensea = "opensea",
   Trader = "trader",
-  Looksrare = "looksrare",
+  LooksRare = "looksrare",
 }
 
-interface Opensea {
+interface ErrorObj {
+  message?: string;
+}
+
+interface ErrorBase {
+  error?: ErrorObj;
+}
+
+interface OpenseaBase extends ErrorBase {
   marketplaceName: MarketplaceName.Opensea;
 }
 
-interface Trader {
+interface TraderBase extends ErrorBase {
   marketplaceName: MarketplaceName.Trader;
 }
 
-interface Looksrare {
-  marketplaceName: MarketplaceName.Looksrare;
+interface LooksRareBase extends ErrorBase {
+  marketplaceName: MarketplaceName.LooksRare;
 }
 
-export interface OpenseaOrder extends Opensea {
-  marketplaceOrder: _OpenseaOrder;
+export interface NormalizedAsset {
```

Comment:
What's the difference between `NormalizedAsset` and `Asset`?

Reply:
`Asset` doesn't allows dumping 4 fields all together, with some of them being optional. You need to write if/else logic, because you can't pass amount for erc721, and you can't pass tokenId for erc20. And on client side you can just ignore empty values, so rendering becomes straightforward, with no conditional logic also.

Reply:
As I was discussing just now, our supported token types are restricted to common intersection of all marketplaces, and in this case limitation is coming from 0x that supports only 3 token types. But orders can be made with more nft types (opensea WyvernSchema got 6 of them), and we should be able to retrieve all of them, so that user could take or cancel them. That's one of the reasons why applying `Asset` type to assets in `NormalizedOrder` is problematic.

Reply:
Also, forgot to mention, `LooksRare` doesn't event returns nft type in order at all. So there is no way to apply `Asset` type to LooksRare case.

Reply:
I'm still for using Asset.
Let's assume we're ok with introducing an unknown asset type which is an intersection of all asset types as a catch-all
```
{
  type: 'unknown';
  contractAddress: string;
  tokenId?: string;
  amount?: bigint;
}
```

## For opensea

`ERC20`, `ERC721`, `ERC1155` are pretty clear cut
`ERC721v3` can be pretty safely assumed to be `ERC721`
`LegacyEnjin`, `ENSShortNameAuction` can be treated as unknown until we figure out what assets fall under them (or just wait for all legacy orders to expire eventually)

OrderV2 will only have these types which are more straightforward:
NATIVE = 0,
ERC20 = 1,
ERC721 = 2,
ERC1155 = 3,
ERC721_WITH_CRITERIA = 4,
ERC1155_WITH_CRITERIA = 5,

## For looksrare

We can either a) classify assets under unknown or b) make api calls looksrare api to retrieve types
I believe we can just go with (a) for now but (b) will work as well if there's the need

## For FE

I understand your argument for the need for FE to just render without caring about types, this can still be achieved by dumping the data into a var with a type that's similar to `unknown`

```typescript
const asset1: Erc721Asset = {
  contractAddress: "0xasdf",
  type: "ERC721",
  tokenId: "asdfa",
};

const asset2: Erc1155Asset = {
  amount: 1n,
  contractAddress: "0xasdf",
  type: "ERC1155",
  tokenId: "12345",
};

interface AssetType {
  amount?: bigint;
  contractAddress: string;
  type: "ERC1155" | "ERC721" | "ERC20";
  tokenId?: string;
}

const assetA: AssetType = asset1;
const assetB: AssetType = asset2;
```

In any case throwing away the typings is NOT the answer even tho it's more convenient for certain cases and for us to implement. Whether the consumer wants to take advantage of the typings we provide is up to them, we should not just take away that choice.

HOWEVER
If you need this to be in quickly to continue your FE work I'm ok with merging the current state of things in. I can make the changes to it later if you need to focus on your FE work, just need to assume on your FE that types will likely happen.